### DISCUSSION TITLE
Deploying using Make

### DISCUSSION THREAD

**Discussion by HenryMiller** [codefori/vscode-ibmi#2397] [Q&A]
👉🏻 Issue text goes here.

Hello,
I am facing a weird behaviour while deploying.
It happens that when I deploy (I always use option "deploy current")   it takes ages to complete the task.
I had a look at my Mac and I saw that the tarball is 790MB.
At 16.08 the tmp files were just a few KB.  At 17.17 I just deployed a "create table "  source member.
It's happening more and more frequently every day .
I had to delete all tmp*  files from var/folders..... and then it works fine for a couple of hours.
Then with no reason it start doing the same thing and I have to close VS Code , delete tmp files and deploy again.
I delete ILEDITOR library as well in order to be 100% sure that next time "deploy" is working fine.

<img width="550" alt="image" src="https://github.com/user-attachments/assets/33d57296-dc23-46f6-ae8c-06c2162bbf47">

⚠️ **REMOVE THIS LINE AND ANY SENSITIVE INFORMATION BELOW!** ⚠️

|Context|Version|
|-|-|
|Code for IBM i version|2.13.6|
|Visual Studio Code version|1.95.3|
|Operating System|darwin_arm64|

<details>
<summary>Active extensions</summary>

```
Code Coverage for IBM i (code-coverage-ibmi): 0.1.7
Code Runner (code-runner): 0.12.2
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.5.0
Configuration Editing (configuration-editing): 1.0.0
Db2 for IBM i (vscode-db2i): 1.6.3
ESLint (vscode-eslint): 3.0.10
Emmet (emmet): 1.0.0
Extension Authoring (extension-editing): 1.0.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
GitLens — Git supercharged (gitlens): 16.0.3
IBM i Debug (ibmidebug): 2.0.1
IBM i Notebooks (vscode-ibmi-notebooks): 0.0.6
IBM i Project Explorer (vscode-ibmi-projectexplorer): 2.12.0
JSON Language Features (json-language-features): 1.0.0
Jira and Bitbucket (Atlassian Labs) (atlascode): 3.0.13
Live Server (LiveServer): 5.7.9
Makefile Tools (makefile-tools): 0.11.13
Merge Conflict (merge-conflict): 1.0.0
NPM support for VS Code (npm): 1.0.1
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Overtype (overtype): 0.5.0
RPGLE (vscode-rpgle): 0.26.14
Redis for VS Code (redis-for-vscode): 1.0.0
Render Line Endings (render-crlf): 1.8.5
Source Orbit (vscode-sourceorbit): 1.0.2
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
XML (vscode-xml): 0.27.1
XML Tools (xml): 2.5.1
YAML (vscode-yaml): 1.15.0
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R4M0|
|Tech Refresh|8|
|CCSID Origin|1144|
|Runtime CCSID|1144|
|Default CCSID|1144|
|SQL|Enabled
|Source dates|Enabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin|/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|/QOpenSys/QIBM/ProdData/JavaVM/jdk80|
|-|-|-|-|-|
|bash|attr|QZDFMDB2.PGM|startDebugService.sh|64bit|
|chsh|iconv||||
|git|setccsid||||
|ls|tar||||
|md5sum|||||
|sort|||||
|stat|||||
</details>

<details>
<summary>Shell env</summary>

```bash
BUILDLIB=M11D00_OBJ
CURLIB=M11D00_OBJ
HOME=/home/HM
HOST=93.42.207.91
LIBLS=BMS74L_VED QDEVTOOLS M11D00_SRC M11D00_SRP M11D00_OBJ M11D00_OBP BOX5DAT M11D00_DBE QTEMP QGPL
LOGIN=hm
LOGNAME=hm
MAIL=/var/spool/mail/hm
OLDPWD=/home/HM
PATH=/QOpenSys/pkgs/bin:/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin:/QOpenSys/usr/bin:/usr/bin
PWD=/home/HM
SHELL=/QOpenSys/pkgs/bin/bash
SHLVL=1
SSH_CLIENT=156.67.23.57 53686 22
SSH_CONNECTION=156.67.23.57 53686 199.5.83.1 22
TZ=<UTC>0
USER=hm
USERNAME=hm
WORKDIR=/home/HM
_=/QOpenSys/pkgs/bin/env
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "£§$"
}
```
</details>

<details>
<summary>Errors</summary>

```json
[
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPD0074:  Il valore 'DOCFUNCTIO' per FROMFILE supera 10 caratteri.\nCPF0001:  Nel comando CPYTOIMPF è stato trovato un errore.",
    "cwd": "/home/HM"
  },
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF4102:  Il file EVFEVENT nella libreria M11D00_OBJ con il membro INSTALL non è stato trovato.\nCPF2803:  Il file di destinazione EVFEVENT in M11D00_OBJ non è consentito.\nCPF2817:  Il comando di copia si è interrotto a causa di un errore.",
    "cwd": "/home/HM"
  },
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF4102:  Il file EVFEVENT nella libreria M11D00_OBJ con il membro INSTALL non è stato trovato.\nCPF2803:  Il file di destinazione EVFEVENT in M11D00_OBJ non è consentito.\nCPF2817:  Il comando di copia si è interrotto a causa di un errore.",
    "cwd": "/home/HM"
  }
]
```
</details>

**Comment by sebjulliand** [comment]
Which one is it you're calling "Deploy current"?
![image](https://github.com/user-attachments/assets/d6893d7e-dfb5-43f4-8192-bb3718d9efb8)

**Comment by HenryMiller** [comment]
Hi, 
I always use this option 
<img width="577" alt="image" src="https://github.com/user-attachments/assets/f7359f34-bdc5-42a2-b70e-a67c7ee80c6b">



and this is what is showing on the bottom bar.



<img width="593" alt="image" src="https://github.com/user-attachments/assets/5750fe35-c930-4b68-ad2e-8b665f126c3d">

**Comment by sebjulliand** [comment]
Sorry, that's not the deployment method you're showing, but the action list. Check out the screenshot I sent you.
If the list I sent doesn't show up for you, did you set a default deployment method in your connection settings?

**Comment by HenryMiller** [comment]
Hi, 

Here is my settings 
<img width="534" alt="image" src="https://github.com/user-attachments/assets/448e12b1-67ce-4709-8799-04a4dab19a37">

I change a source file , click CMD+E and it shows  
<img width="436" alt="image" src="https://github.com/user-attachments/assets/fcece6c4-acf4-4e0d-bd43-300aa01a2a52">



By the way , 
deploying is still there....

<img width="385" alt="image" src="https://github.com/user-attachments/assets/32144963-f7f8-40a7-947e-18743ef66a30">

**Comment by sebjulliand** [comment]
The `Changes` deployment method gathers all the files changed since the last deployment (or since VS Code was started, whichever comes first), based on a VS Code listener that listens to changes made in the current workspace.

It could be interesting to temporarily change your default deployment method to No Default:
![image](https://github.com/user-attachments/assets/c440e023-72e6-41cc-b2a1-d5de09c3e07e)

Then, whenever you'll run your action, you'll be asked to choose a deployment method. The `Change` item in the selection box will display files were detected as changed.
![image](https://github.com/user-attachments/assets/317998d5-e2e7-4a2f-aaa0-3ffb4b75fa6c)
You'll be able to check beforehand if the changed files count seems to be crazy.

I'd like to check what happens before the listener goes crazy and thinks everything was changed. Do you remember doing something along these lines before it goes crazy?
- Disconnecting and reconnecting multiple times
- Changing your workspace folder
- Reloading VS Code
- Reloading extensions

Let me know if you can think of anything.
Thanks!

**Comment by HenryMiller** [comment]
Hello,
I deployed  until yesterday evening with no problem whatsoever.
This morning at first deploy I got this message.

The last thing  I did yesterday before shutting down my Mac was "ng build" for the Angular FrontEnd 

<img width="451" alt="image" src="https://github.com/user-attachments/assets/623d4d41-f676-4321-81b1-9958205b48ba">

**Comment by sebjulliand** [comment]
Does `ng build` actually changes any file in your workspace?
Does your workspace only contains IBM i source files or does it includes both the backend and frontend sources?

**Comment by HenryMiller** [comment]
It includes both.

**Comment by sebjulliand** [comment]
There is a very good chance that "it works as expected" then...in other words, it's not a bug, it's a feature 😅 

I assume that running `ng build` will make some changes in the workspace that are getting caught by the listener. So deploying  will include all these changed files (the listener doesn't pickup changes made only to IBM i related files).

The best thing for you to do, provided that your backend and frontend files are in two separate root folders, would be to open both folders as workspace folders.

To do so, you would first need to open VS Code on the backend root folder, and then use `Add folder to workspace` to add the frontend folder to your workspace.
![image](https://github.com/user-attachments/assets/3e32cc26-0041-4556-a3da-8f19a9132cf8)

Since code for IBM i treats each workspace folder separately, you'll be able to deploy only what's been changed in the backend folder using this methodology.

Also, once your workspace is correctly set up, you can save it to re-open it more easily.
![image](https://github.com/user-attachments/assets/f7446951-59a6-44b8-a228-6c34c23e6f7a)

**Comment by HenryMiller** [comment]
Hello,
thank you!
I will dig into it because all changed files are in the same workspace but they are also in .gitignore file  and they have not been deployed to Ibm I.

**Comment by HenryMiller** [comment]
Hello,
It seems the issue still persists.
I worked as usual , but suddenly I got the same deployment issue.

<img width="711" alt="image" src="https://github.com/user-attachments/assets/0a3dd233-51d2-48ee-891c-31f6565acc97">

Closing VS Code and reopening it always  shows a strange message regarding a stmf members  that I never changed since first deployment .
<img width="541" alt="image" src="https://github.com/user-attachments/assets/1addbdfa-e574-4055-b38e-48b0cf517564">

As I'm usually working over a VPN connection, could the problem be caused by a temporary disconnection or something like that ?
is there any option to kill  the deployment task.
The telnet session hasn't dropped and it is working fine.

I disconnected from VPN , closed VS Code , reconnected to VPN , reopened Vs Code  and then deployed successfully  in 10 seconds.


Thanks
